lang/java/reef-io/src/main/java/org/apache/reef/io/network/naming/NameServerImpl.java

```diff
@@ -63,14 +61,13 @@
   /**
    * @param port    a listening port number
    * @param factory an identifier factory
-   * @deprecated inject the NameServer instead of new it up
+   * @param localAddressProvider a local address provider
    * Constructs a name server
    */
-  // TODO: All existing NameServer usage is currently new-up, need to make them injected as well.
-  @Deprecated
-  public NameServerImpl(
-      final int port,
-      final IdentifierFactory factory,
+  @Inject
+  private NameServerImpl(
```

Comment:
Please leave this `public`. We don't know who uses it, and making it `private` would break them. Also, please add a deprecation and in the JavaDoc indicate when it was deprecated, like so:

```
  @deprecated in 0.12. Use Tang to obtain an instance of this or, better, NameServer, instead.
```